test/integration/model/scope/associations.test.js

```diff
@@ -418,6 +418,31 @@ describe(Support.getTestDialectTeaser('Model'), () => {
               expect(user.dataValues).not.to.have.property('secret');
             });
         });
+
+        it('should not throw error', function() {
+          const Clientfile = this.sequelize.define('clientfile');
+          const Mission = this.sequelize.define('mission', { secret: Sequelize.STRING });
+          const Building = this.sequelize.define('building');
+          Clientfile.hasOne(Mission);
+          Clientfile.hasOne(Building);
+
+          return this.sequelize.sync({ force: true })
+            .then(() => {
+              return Clientfile.findAll({
+                include: [
+                  {
+                    association: 'mission',
+                    where: {
+                      secret: 'foo'
+                    }
+                  },
+                  {
+                    association: 'building'
```

Comment:
What if you are passing association object, then `uniqueInclude` can fail due to `[object]` stringification